File: CppHW/Week09/Ticket.cpp
#include "Ticket.h"
#include <iostream>
using namespace std;

Ticket::Ticket(): Ticket(0.0)
{
}

Ticket::Ticket(const int& price)
	:price(price)
{
	static int n = 1;
	this->number = n++;
}

Ticket::~Ticket()
{
}

int Ticket::getNumber() const
{
	return this->number;
}

double Ticket::getPrice() const
{
	return this->price;
}

void Ticket::setPrice(const double& price)
{
	this->price = price;
}

void Ticket::show() const
{
	cout << "티켓번호: " << this->number << endl;
	cout << "가격정보: " << this->price << endl;
}

File: JavaLecture_src/Week05lab/Ticket.java
package Week05lab;

public abstract class Ticket {
	protected int number;
	protected double price;
	
	public Ticket(int number, double price) {
		this.number = number;
		this.price = price;
	}

	public Ticket(int number) {
		this(number, 0.0);
		this.number = number;
	}
	

	public Ticket() {
		this(0,0.0);
	}
	
	
	public int getNumber() {
		return number;
	}


	public void setNumber(int number) {
		this.number = number;
	}

//	public abstract double getPrice();

	public double getPrice() {
		return price;
	}


	public void setPrice(double price) {
		this.price = price;
	}


	@Override
	public String toString() {
		return"티켓번호: "+this.number+"\n"+"티켓가격: " +this.price;
	}
}
